### PR TITLE
用 Python 编程进行糖尿病相关的机器学习

### LINKED ISSUE #3610: 用 Python 编程进行糖尿病相关的机器学习
* 原文链接：[PROGRAMMING Machine Learning for Diabetes with Python](https://datascienceplus.com/machine-learning-for-diabetes-with-python/)
  * Markdown文件：[文件地址](https://github.com/xitu/gold-miner/blob/master/TODO1/machine-learning-for-diabetes-with-python.md)
  * PR 地址：#3622
  * 文章分类：**机器学习**
  * **注意：文件位置在 `TODO1` 文件夹中，不是之前的 `TODO` 了。**
  ----
  * 翻译时间：3 天
  * 校对时间：1 天
  * 翻译积分：3 分
  * 校对积分：1 分
  * [积分有什么用](https://github.com/xitu/gold-miner/wiki)

### PR DESCRIPTION
**Pull request by EmilyQiRabbit**
译文翻译完成，resolve #3610

### CODE REVIEW

**Comment by EmilyQiRabbit** [comment]
@leviding 翻译完成

**Comment by leviding** [comment]
@EmilyQiRabbit 把这个 PR 的名字改成本文的中文标题

**Comment by luochen1992** [comment]
校对认领

**Comment by fanyijihua** [comment]
@luochen1992 好的呢 :beer:

File: TODO1/machine-learning-for-diabetes-with-python.md [hunk 1]
```diff
@@ -278,21 +278,21 @@ Accuracy on training set: 0.800
...
 plot_feature_importances_diabetes(rf)
 ```
 
-Gives this plot:
+得出下图：
 
 ![]=(https://datascienceplus.com/wp-content/uploads/2018/03/diabetes_8.png)
 
-Similarly to the single decision tree, the random forest also gives a lot of importance to the “Glucose” feature, but it also chooses “BMI” to be the 2nd most informative feature overall. The randomness in building the random forest forces the algorithm to consider many possible explanations, the result being that the random forest captures a much broader picture of the data than a single tree.
+和单一决策树相似，随机森林的 “Glucose” 特征权重也比较高，但是还选出了 “BMI” 作为所有特征中第二高的权重。生成随机森林时的随机性要求算法必须考虑众多可能的解答，结果就是随机森林比单一决策树能够更完整的捕捉到数据的特征。
```
Comment (luochen1992) [inline TODO1/machine-learning-for-diabetes-with-python.md]: 『更完整的捕捉』=>『更完整地捕捉』

File: TODO1/machine-learning-for-diabetes-with-python.md [hunk 2]
```diff
@@ -64,27 +64,27 @@ diabetes.info()
 
 ![](https://datascienceplus.com/wp-content/uploads/2018/03/diabetes_4.png)
 
-## k-Nearest Neighbors
+## k 近邻
 
-The k-NN algorithm is arguably the simplest machine learning algorithm. Building the model consists only of storing the training data set. To make a prediction for a new data point, the algorithm finds the closest data points in the training data set — its “nearest neighbors.”
+k 近邻算法可以说是最简单的机器学习算法。它建立仅包含训练数据集的模型。为了对一个新的数据点作出预测，算法将在训练数据集中找到最近的数据点 - 它的 “最近邻点”。
```
Comment (luochen1992) [inline TODO1/machine-learning-for-diabetes-with-python.md]: [全角标点与其他字符之间不加空格](https://github.com/xitu/gold-miner/wiki/%E8%AF%91%E6%96%87%E6%8E%92%E7%89%88%E8%A7%84%E5%88%99%E6%8C%87%E5%8C%97#%E5%85%A8%E8%A7%92%E6%A0%87%E7%82%B9%E4%B8%8E%E5%85%B6%E4%BB%96%E5%AD%97%E7%AC%A6%E4%B9%8B%E9%97%B4%E4%B8%8D%E5%8A%A0%E7%A9%BA%E6%A0%BC)

File: TODO1/machine-learning-for-diabetes-with-python.md [hunk 3]
```diff
@@ -382,7 +382,7 @@ Accuracy on training set: 0.77
 Accuracy on test set: 0.77
 ```
 
-Scaling the data made a huge difference! Now we are actually underfitting, where training and test set performance are quite similar but less close to 100% accuracy. From here, we can try increasing either C or gamma to fit a more complex model.
+数据归一化导致了巨大的不同！现在拟合度正好，训练集和测试集的表现相似但是距离 100% 的精度还有点远。此时，我们可以试着提高 C 或者 gamma 来生成一个更复杂的模型。
```
Comment (luochen1992) [inline TODO1/machine-learning-for-diabetes-with-python.md]: 〖现在拟合度正好』=>〖现在欠拟合了』
从原文和实验结果来看都是是欠拟合，后文也解释了 —— 『离100%有点远』。

File: TODO1/machine-learning-for-diabetes-with-python.md [hunk 4]
```diff
@@ -414,7 +414,7 @@ Accuracy on training set: 0.71
 Accuracy on test set: 0.67
 ```
 
-The accuracy of the Multilayer perceptrons (MLP) is not as good as the other models at all, this is likely due to scaling of the data. deep learning algorithms also expect all input features to vary in a similar way, and ideally to have a mean of 0, and a variance of 1\. We must re-scale our data so that it fulfills these requirements.
+多层感知（Multilayer perceptrons）的精度远不如其他模型的好，这可能是因为数据的量纲。深度学习算法同样希望所有输入特征归一化，并且最好均值为 0，方差为 1。我们必须重新调整数据，让它满足这些要求。
```
Comment (luochen1992) [inline TODO1/machine-learning-for-diabetes-with-python.md]: 〖多层感知（Multilayer perceptrons）』=>〖多层感知器（Multilayer perceptrons）』

File: TODO1/machine-learning-for-diabetes-with-python.md [hunk 5]
```diff
@@ -480,21 +480,21 @@ plt.ylabel("Input feature")
 plt.colorbar()
 ```
 
-Gives this plot:
+得出下图：
 
 ![](https://datascienceplus.com/wp-content/uploads/2018/03/diabetes_10.png)
 
-From the heat map, it is not easy to point out quickly that which feature (features) have relatively low weights compared to the other features.
+从热图上不难很快就看出，相比去其他特征，那些特征的权重比较低。
```
Comment (luochen1992) [inline TODO1/machine-learning-for-diabetes-with-python.md]: 〖相比去其他特征』=>〖相比其他特征』

Comment (luochen1992) [reply TODO1/machine-learning-for-diabetes-with-python.md]: 看图也能明白，很难看出哪些特征重要，哪些不重要。
所以该句应该是：
〖从热图上不难很快就看出，相比去其他特征，那些特征的权重比较低。』=>〖从热图上，很难快速指出哪些特征与其他特征相比具有相对较低的权重。』

File: TODO1/machine-learning-for-diabetes-with-python.md [hunk 6]
```diff
@@ -480,21 +480,21 @@ plt.ylabel("Input feature")
...
 
-## Summary
+## 总结
 
-We practiced a wide array of machine learning models for classification and regression, what their advantages and disadvantages are, and how to control model complexity for each of them. We saw that for many of the algorithms, setting the right parameters is important for good performance.
+为了分类和回归，我们试验了各种各样的机器学习模型，（知道了）它们的优点和缺点都是什么，以及如何控制每个模型的复杂度。我们发现对于很多算法，设置合适的参数对于模型的上佳表现至关重要。
 
-We should be able to know how to apply, tune, and analyze the models we practiced above. It’s your turn now! Try applying any of these algorithms to the built-in datasets in [scikit-learn](https://datascienceplus.com/multi-class-text-classification-with-scikit-learn/) or any data set at your choice. Happy Machine Learning!
+我们应该知道了如何应用，调参，并分析上文中我们试验过的模型。现在轮到你了！试着在 [scikit-learn](https://datascienceplus.com/multi-class-text-classification-with-scikit-learn/) 的内置数据集或者其他你选择的任意数据集上应用这些算法中的任一一个。快乐的进行机器学习吧！
```
Comment (luochen1992) [inline TODO1/machine-learning-for-diabetes-with-python.md]: 〖应用，调参，并』=>〖应用、调参并』

**Comment by luochen1992** [comment]
@EmilyQiRabbit @leviding 校对完成

**Comment by zhmhhu** [comment]
校对认领 @leviding

**Comment by fanyijihua** [comment]
@zhmhhu 妥妥哒 :beers:

File: TODO1/machine-learning-for-diabetes-with-python.md [hunk 7]
```diff
@@ -156,13 +156,13 @@ Training set accuracy: 0.785
...
+使用 C=100 导致在训练集上的精度略有上升但是在测试集的精度下降，我们可以确定低正则和更复杂的模型也许并不能比默认设置表现更好。
 
-Therefore, we should choose default value C=1.
+因此我们应该采用默认值 C=1。
 
-Let’s visualize the coefficients learned by the models with the three different settings of the regularization parameter C.
+我们来将模型通过学习三个不同正则化参数 C 的数据集所得到的参数可视化。
 
-Stronger regularization (C=0.001) pushes coefficients more and more toward zero. Inspecting the plot more closely, we can also see that feature “DiabetesPedigreeFunction”, for C=100, C=1 and C=0.001, the coefficient is positive. This indicates that high “DiabetesPedigreeFunction” feature is related to a sample being “diabetes”, regardless which model we look at.
+正则化比较强（C=0.001）的集合得到的参数越来越靠近零。更仔细的看图，我们也能发现，对于 C=100，C=1 和 C=0.001，特征 “DiabetesPedigreeFunction” 系数都是正值。这意味着，不管哪个模型，高 “DiabetesPedigreeFunction” 特征和样本是糖尿病是相关联的。
```
Comment (zhmhhu) [inline TODO1/machine-learning-for-diabetes-with-python.md]: “这意味着，不管哪个模型，高 “DiabetesPedigreeFunction” 特征和样本是糖尿病是相关联的” =>
 "这意味着，不管我们看的是哪个模型，高 “DiabetesPedigreeFunction” 特征和糖尿病样本是相关联的"

File: TODO1/machine-learning-for-diabetes-with-python.md [hunk 8]
```diff
@@ -156,13 +156,13 @@ Training set accuracy: 0.785
...
 ```
 
-Using C=100 results in a little bit higher accuracy on the training set and little bit lower accuracy on the test set, confirming that less regularization and a more complex model may not generalize better than default setting.
+使用 C=100 导致在训练集上的精度略有上升但是在测试集的精度下降，我们可以确定低正则和更复杂的模型也许并不能比默认设置表现更好。
 
-Therefore, we should choose default value C=1.
+因此我们应该采用默认值 C=1。
 
-Let’s visualize the coefficients learned by the models with the three different settings of the regularization parameter C.
+我们来将模型通过学习三个不同正则化参数 C 的数据集所得到的参数可视化。
```
Comment (zhmhhu) [inline TODO1/machine-learning-for-diabetes-with-python.md]: "我们来将模型通过学习三个不同正则化参数 C 的数据集所得到的参数可视化"  => "我们来将通过对三个不同正则化参数 C 的数据集建立的模型进行学习所得到的参数可视化" 或者 “我们参数可视化，这些参数是通过学习对三个不同正则化参数 C 的数据集建立的模型所得到的”

**Comment by EmilyQiRabbit** [comment]
@leviding 校对完成

**Comment by EmilyQiRabbit** [comment]
@zhmhhu @luochen1992 谢谢两位～

**Comment by leviding** [comment]
@EmilyQiRabbit 已经 merge 啦~ 快快麻溜发布到[掘金](https://juejin.im)然后给我发下链接，方便及时添加积分哟。

掘金翻译计划有自己的知乎专栏，你也可以投稿哈，推荐使用一个[好用的插件](https://chrome.google.com/webstore/detail/minimalist-markdown-edito/pghodfjepegmciihfhdipmimghiakcjf?hl=zh-CN)。
专栏地址：https://zhuanlan.zhihu.com/juejinfanyi
